File: f/parallel/src/mat_con1.f
!C
!C***
!C*** MAT_CON1
!C***
!C
      subroutine MAT_CON1
      use pfem_util
      implicit REAL*8 (A-H,O-Z)

      allocate (indexL(0:NP), indexU(0:NP))
      indexL= 0
      indexU= 0

      do i= 1, NP
        indexL(i)= indexL(i-1) + INL(i)
        indexU(i)= indexU(i-1) + INU(i)
      enddo

      NPL= indexL(NP)
      NPU= indexU(NP)

      allocate (itemL(NPL), itemU(NPU))

      do i= 1, NP
        do k= 1, INL(i)
                kk = k + indexL(i-1)
          itemL(kk)=        IAL(i,k)
        enddo
        do k= 1, INU(i)
                kk = k + indexU(i-1)
          itemU(kk)=        IAU(i,k)
        enddo
      enddo

      deallocate (INL, INU, IAL, IAU)

      end subroutine MAT_CON1

File: c/parallel/src/mat_con1.c
/***
 *** MAT_CON1
 ***/
#include <stdio.h>
#include "pfem_util.h"
#include "allocate.h"
extern FILE* fp_log;
void MAT_CON1()
{
	int i,k,kk;

	indexL=(KINT*)allocate_vector(sizeof(KINT),NP+1);
	indexU=(KINT*)allocate_vector(sizeof(KINT),NP+1);
	for(i=0;i<NP+1;i++) indexL[i]=0;
	for(i=0;i<NP+1;i++) indexU[i]=0;

	for(i=0;i<NP;i++){
		indexL[i+1]=indexL[i]+INL[i];
		indexU[i+1]=indexU[i]+INU[i];
	}

	NPL=indexL[NP];
	NPU=indexU[NP];

	itemL=(KINT*)allocate_vector(sizeof(KINT),NPL);
	itemU=(KINT*)allocate_vector(sizeof(KINT),NPU);

	for(i=0;i<NP;i++){
		for(k=0;k<INL[i];k++){
			kk=k+indexL[i];
			itemL[kk]=IAL[i][k];
		}
		for(k=0;k<INU[i];k++){
			kk=k+indexU[i];
			itemU[kk]=IAU[i][k];
		}
	}
 

	deallocate_vector(INL);
	deallocate_vector(INU);
	deallocate_vector(IAL);
	deallocate_vector(IAU);
}
